sdk/src/Core/Internal/Utils/DaemonConfig.cs

```diff
@@ -48,29 +48,44 @@ public class DaemonConfig
         public static readonly IPEndPoint DefaultEndpoint = new IPEndPoint(_defaultDaemonAddress, _defaultDaemonPort);
 
         /// <summary>
-        /// Gets aor sets UDP endpoint.
+        /// Gets or sets UDP endpoint.
         /// </summary>
-        public IPEndPoint UDPEndpoint { get; set; }
+        public EndPoint UDPEndpoint { get; set; }
 
         /// <summary>
         /// Gets or sets TCP endpoint.
         /// </summary>
-        public IPEndPoint TCPEndpoint { get; set; }
+        public EndPoint TCPEndpoint { get; set; }
+        
+        /// <summary>
+        /// Gets IP for UDP endpoint.
+        /// </summary>
+        public IPEndPoint UDP_IP_Endpoint {

```

Comment:
Instead of adding new property for UDP and TCP, we can have internal variables `_udpEndpoint` and `_tcpEndpoint` of type `Endpoint`.
As a result, the original method `public IPEndPoint UDPEndpoint { get; set; }` will look like `public IPEndPoint UDPEndpoint { 
     get {_udpEndpoint.GetIPEndpoint()}
 }`

I think when users create instance of `DaemonConfig` the only public property should be `UDPEndpoint` and `TCPEndpoint` of type `IPEndPoint`. The getters on these should take care of returning correct IP Address.

The additional properties `TCP_IP_Endpoint` and `UDP_IP_Endpoint` may confuse the users. Please let me know your thoughts.

Reply:
I agree that the interface might be a bit clunky. The drawback of ```public IPEndPoint UDPEndpoint { get {_udpEndpoint.GetIPEndpoint()} }```is that the endpoint property becomes immutable from an outside perspective. But that might be okay?

Reply:
I think that 36c1c83 simplifies that interface as requested.